setup/mail-dovecot.sh

```diff
@@ -46,8 +46,45 @@ tools/editconf.py /etc/dovecot/conf.d/10-master.conf \
 # See http://www.dovecot.org/pipermail/dovecot/2013-March/088834.html.
 # A reboot is required for this to take effect (which we don't do as
 # as a part of setup). Test with `cat /proc/sys/fs/inotify/max_user_instances`.
+# Added IP Spoofing detection and other settings to harden network 
+# by preventing source routing of incoming packets and log malformed IP's 
 tools/editconf.py /etc/sysctl.conf \
 	fs.inotify.max_user_instances=1024
+	# IP Spoofing protection
+	net.ipv4.conf.all.rp_filter = 1
+	net.ipv4.conf.default.rp_filter = 1
+
+	# Ignore ICMP broadcast requests
+	net.ipv4.icmp_echo_ignore_broadcasts = 1
+
+	# Disable source packet routing
+	net.ipv4.conf.all.accept_source_route = 0
+	net.ipv6.conf.all.accept_source_route = 0 
+	net.ipv4.conf.default.accept_source_route = 0
+	net.ipv6.conf.default.accept_source_route = 0
+
+	# Ignore send redirects
+	net.ipv4.conf.all.send_redirects = 0
+	net.ipv4.conf.default.send_redirects = 0
+	
+	# Block SYN attacks
+	net.ipv4.tcp_syncookies = 1
+	net.ipv4.tcp_max_syn_backlog = 2048
+	net.ipv4.tcp_synack_retries = 2
+	net.ipv4.tcp_syn_retries = 5
+
+	# Log Martians
+	net.ipv4.conf.all.log_martians = 1
+	net.ipv4.icmp_ignore_bogus_error_responses = 1
+
+	# Ignore ICMP redirects
+	net.ipv4.conf.all.accept_redirects = 0
+	net.ipv6.conf.all.accept_redirects = 0
+	net.ipv4.conf.default.accept_redirects = 0 
+	net.ipv6.conf.default.accept_redirects = 0
+
+	# Ignore Directed pings
+	net.ipv4.icmp_echo_ignore_all = 1
```

Comment:
Same comment as for not answering to broadcast ping. `fail2ban` is protecting the system against abuse. And given that this is a *public* mail machine, this IP address is advertised, and will respond to `syn/ack`s so there's little use to hide from ICMP requests.